tests/unit/transformers/test_categorical.py

```diff
@@ -156,7 +156,7 @@ def test__get_intervals_nans(self):
             categorical value (start, end).
         """
         # Setup
-        data = pd.Series(['foo', np.nan, None, 'foo', 'foo', 'tar'])
```

Comment:
Can we change the expected value instead of this part? I think it's good to know how this function handles `None`

Reply:
We can't really do it, since there are two distinct behaviors for this input, depending on what version of pandas we are running. For `pandas 1.4.0`, the existing code doesn't even work on this input. Notice that the categorical transformer does the following:

https://github.com/sdv-dev/RDT/blob/40d2667cab950164916e7ac58a317621f99afce8/rdt/transformers/categorical.py#L106-L109

So we actually just overwrite the first `np.nan` with the `None`, which is not the intended behavior. If we do want to keep support for multiple types of nan's, we could cast all null objects no `np.nan` right at the start of the function.

Reply:
It's weird that `value__counts` doesn't treat `None` as `np.nan`, but `pd.isna` will still return `True` for it. My question is, is it possible for `_get_intervals` to receive data with different null types. If so, we have to handle that which means I think we should convert them all before doing the `value_counts`